### PR TITLE
Update opentelemetry deps to 0.6.0

### PR DESCRIPTION
**Pull request by ricardoccpaiva**

### CODE REVIEW

**Comment by ricardoccpaiva** [comment]
Any idea of why CI is failing?
````bash
  Package libwxgtk3.0-0v5 is not available, but is referred to by another package.
  This may mean that the package is missing, has been obsoleted, or
  is only available from another source
  
  E: Package 'libwxgtk3.0-0v5' has no installation candidate
  Error: The process '/home/runner/work/_actions/actions/setup-elixir/v1.0.0/src/install-otp-ubuntu' failed with exit code 100

**Comment by tsloughter** [comment]
@ricardoccpaiva not sure why but try upgrading the action maybe https://github.com/opentelemetry-beam/opentelemetry_ecto/blob/master/.github/workflows/test.yml#L26

think you can try `@master`, ont sure what the versions it has released are.

**Comment by bryannaegele** [comment]
The actions have to be changed @tsloughter. They're managed by erlef now. I'll update shortly.

@ricardoccpaiva - same note as on the phoenix library. Can you confirm there are no changes that need to be made for 0.6.0 compliance?

**Comment by garthk** [comment]
@bryannaegele what's necessary to be “compliant”? I'll trust I'll be facing the same question when I fix `opentelemetry_plug` up, and would love to be well prepared.

I'd strongly prefer any integration to delay publishing an 0.6 version until it matches the rest of the packages with 0.6 versions. It's hard to figure that out, though, without being able to add a dependency to it. To break the tie, are we OK merging a dependency bump like this one as long as it uses an RC version `0.6.0-rc.1` while we decide whether or not it does match the others?

**Comment by bryannaegele** [comment]
> what's necessary to be “compliant”?

We have to match any changes in the trace API (and soon metrics) parts of the spec, semantics sections in the spec (database in this case plus any changes to common stuff, such as trace span), and anything that would go into the protos. The otel api/sdk versions are all tied to the protos and the libraries, such that v0.6 of the protobufs, API, and SDK implementations all comply with v0.6 of the specification.

The changelogs are really hard to follow, so I usually go to all of those parts of the specification and do diffs between the tags on those files and see what needs changing. `git diff v0.5.0 v0.6.0 specification/trace/semantic_conventions/database.md`. In general for these bridge libraries, we're currently concerned with meeting all `required` in the semantics and doing any optionals we can which don't pose a significant performance concern OR should probably be made optional through configuration at some point. The latter two are judgement calls for the time being.

**Comment by ricardoccpaiva** [comment]
will this PR be closed in favor of #12 ?

**Comment by bryannaegele** [comment]
@ricardoccpaiva yeah. I think we're just going to move everything to 1.0-rc instead of trying to cover all the intermediaries. That will leave us with just needing to mark it all 1.0 once we get out of the release candidates for otel.
